src/ImpurityDensities/PtypePNJunctionImpurityDensity.jl

```diff
@@ -0,0 +1,55 @@
+"""
+    struct ThermalDiffusionLithiumDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+
+A PN junction impurity model based on lithium thermal diffusion and constant bulk impurity density.
+The surface lithium density is saturated.
+
+ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+ 
+## Fields
+* `lithium_annealing_temperature::T`: lithium annealing temperature when the lithium is diffused into the crystal. The default value is 623 K.
+* `lithium_annealing_time::T`: lithium annealing time. The default value is 18 minutes.
+* `surface_imp_model::ThermalDiffusionLithiumDensity{T,G}`: the density profile of lithium (n-type). This will be constructed with above parameters.
+* `bulk_imp_model::AbstractImpurityDensity{T}`: the density of the p-type impurity.
+"""
+
+struct PtypePNJunctionImpurityDensity{T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}} <: AbstractImpurityDensity{T}
+	lithium_annealing_temperature::T
+	lithium_annealing_time::T
+    contact_with_lithium_doped::G
+    surface_imp_model::ThermalDiffusionLithiumDensity{T,G}
+    bulk_imp_model::AbstractImpurityDensity{T}
+end
```

Comment:
I think this could be made more generic by removing everything lithium-related (the lithium part is anyhow saved in `surface_imp_model`)
```suggestion
struct PtypePNJunctionImpurityDensity{T <: SSDFloat, G <: Union{<:AbstractGeometry, Nothing}} <: AbstractImpurityDensity{T}
    contact_doped::G
    surface_imp_model::ThermalDiffusionLithiumDensity{T,G}
    bulk_imp_model::AbstractImpurityDensity{T}
end
```